api-gateway/src/main/resources/static/scripts/visit-list/visit-list.controller.js

```diff
@@ -19,9 +22,25 @@ angular.module('visitList')
         eventSource.addEventListener('message', function (event){
             $scope.$apply(function(){
                 console.log(event.data)
-                self.upcomingVisits.push(JSON.parse(event.data))
-            })
-        })
+                const visitData = JSON.parse(event.data);
+                switch (visitData.status) {
+                    case "UPCOMING":
+                        self.upcomingVisits.push(visitData);
```

Comment:
Sorting was not modified for the new lists. If you select any sorting button on any table, it will only modify the sorting for upcoming visits.

Reply:
I see, I'll speak to you in person to make sure my understanding of the problem is correct